File: backend/src/models/balance_variant.js
const Sequelize = require('sequelize');
module.exports = function(sequelize, DataTypes) {
  return sequelize.define('balance_variant', {
    id: {
      autoIncrement: true,
      type: DataTypes.INTEGER,
      allowNull: false,
      primaryKey: true
    },
    note: {
      type: DataTypes.STRING(255),
      allowNull: true
    },
    real_quantity: {
      type: DataTypes.INTEGER,
      allowNull: false
    },
    saved_quantity: {
      type: DataTypes.INTEGER,
      allowNull: false
    },
    variant_id: {
      type: DataTypes.INTEGER,
      allowNull: true,
      references: {
        model: 'variant',
        key: 'id'
      }
    },
    balance_id: {
      type: DataTypes.INTEGER,
      allowNull: true,
      references: {
        model: 'warehouse_balance',
        key: 'id'
      }
    }
  }, {
    sequelize,
    tableName: 'balance_variant',
    timestamps: true,
    indexes: [
      {
        name: "PRIMARY",
        unique: true,
        using: "BTREE",
        fields: [
          { name: "id" },
        ]
      },
      {
        name: "FKggrctc42mw5do93o7t7wh4ahn",
        using: "BTREE",
        fields: [
          { name: "variant_id" },
        ]
      },
      {
        name: "FK2h24nth1v3f6q3aolm5ytqhd2",
        using: "BTREE",
        fields: [
          { name: "balance_id" },
        ]
      },
    ]
  });
};

File: backend/src/main/java/com/projectcnw/salesmanagement/models/BalanceVariant.java
package com.projectcnw.salesmanagement.models;

import jakarta.persistence.Entity;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.Getter;
import lombok.Setter;

@Entity
@Getter
@Setter
public class BalanceVariant extends BaseEntity {

    private int realQuantity;

    private int savedQuantity;

    private String note;

    @ManyToOne
    @JoinColumn(name = "balance_id")
    private WarehouseBalance warehouseBalance;

    @ManyToOne
    @JoinColumn(name = "variant_id")
    private Variant variant;
}
